javascript/ql/src/Security/CWE-078/IndirectCommandInjection.ql

```diff
@@ -0,0 +1,38 @@
+/**
+ * @name Indirect uncontrolled command line
+ * @description Using command line parameters to construct other
+ *              command line arguments may allow a malicious user to
+ *              indirectly change the meaning of the command.
+ * @kind path-problem
+ * @problem.severity warning
+ * @precision medium
+ * @id js/indirect-command-line-injection
+ * @tags correctness
+ *       security
+ *       external/cwe/cwe-078
+ *       external/cwe/cwe-088
+ */
+
+import javascript
+import DataFlow::PathGraph
+import semmle.javascript.security.dataflow.IndirectCommandInjection::IndirectCommandInjection
+
+/**
+ * Holds if `result` is `src` or a slice thereof.
+ */
+DataFlow::SourceNode getACommandLineArgumentsArray(CommandLineArgumentsArray src) {
+  src = result
+  or
+  getACommandLineArgumentsArray(src).getAMethodCall("slice") = result
+}
+
+from Configuration cfg, DataFlow::PathNode source, DataFlow::PathNode sink, DataFlow::Node highlight
+where
+  cfg.hasFlowPath(source, sink) and
+  // avoid flagging trivial CLI utilities by excluding simple, direct flow from command line arguments to sinks
+  not getACommandLineArgumentsArray(source.getNode()).getAPropertyRead().flowsTo(sink.getNode()) and
+  if cfg.isSinkWithHighlight(sink.getNode(), _)
+  then cfg.isSinkWithHighlight(sink.getNode(), highlight)
+  else highlight = sink.getNode()
+select highlight, source, sink, "This command depends on an unsanitized $@.", source.getNode(),
+  "command-line parameter"
```

Comment:
It may be worth using either "command-line _argument_" or "command-line _parameter_" throughout for consistency.

Reply:
Done. Although a context-dependent choice between "parameter" and "argument" could be more technically correct. For this query however, arguments becomes parameters, which in turn becomes  arguments, so I agree that we can stick to just one in the name of simplicity.